web/src/main/java/ua/com/vertex/controllers/CourseDetailsController.java

```diff
@@ -0,0 +1,124 @@
+    }
+
+    @PostMapping(value = "/searchCourse")
+    public String searchCourse(@Validated @ModelAttribute(COURSE_DATA) Course course,
+                               BindingResult bindingResult, Model model) {
+        LOGGER.debug(String.format("Search user by name - (%s) and finished - (%s).",
+                course.getName(), course.isFinished()));
+        String result = SEARCH_COURSE_JSP;
+
+        if (!bindingResult.hasErrors()) {
+            try {
+                List<Course> courses = courseLogic.searchCourseByNameAndStatus(course);
+                if (courses.isEmpty()) {
+                    model.addAttribute(MSG, String.format("Course with name - '(%s)' not found. " +
+                            "Please check the data and try it again.", course.getName()));
+                } else {
+                    model.addAttribute("courses", courses);
+                }
+            } catch (DataAccessException e) {
+                LOGGER.warn(e);
+                model.addAttribute(MSG, LOGGER_SERVER_EXCEPTION);
+            } catch (Exception e) {
+                LOGGER.warn(e);
+                result = ERROR;
+            }
+        }else {
+            model.addAttribute(MSG, "Enter the correct data !!! ");
+        }
+
+        return result;
+    }
+
+    @PostMapping(value = "/courseDetails")
+    public ModelAndView courseDetails(@RequestParam(COURSE_ID) int courseId) {
+        LOGGER.debug(String.format("Go to the course information page. Course ID -: - (%s)", courseId));
+
+        ModelAndView result = new ModelAndView(COURSE_DETAILS_JSP);
+
+        try {
+            result.addObject(COURSE, courseLogic.getCourseById(courseId).orElseThrow((Exception::new)));
+            result.addObject(TEACHERS, userLogic.getTeachers());
+        } catch (Exception e) {
+            LOGGER.warn(e);
+            result.setViewName(ERROR);
+        }
+
+        return result;
+    }
+
+    @PostMapping(value = "/updateCourse")
+    public String updateCourse(@Valid @ModelAttribute(COURSE) Course course, BindingResult bindingResult, Model model) {
+        LOGGER.debug(String.format("Update course with course ID - (%s). Course details: - (%s)", course.getId(), course));
+        String result = ADMIN_JSP;
+        if (!bindingResult.hasErrors()) {
+            try {
+                model.addAttribute(MSG, String.format("Course with id - (%s) updated!!!",
+                        courseLogic.updateCourseExceptPrice(course)));
+            } catch (DataAccessException e) {
+                LOGGER.warn(e);
+                model.addAttribute(MSG, LOGGER_SERVER_EXCEPTION);
+                result = SEARCH_COURSE_JSP;
+            } catch (Exception e) {
+                LOGGER.warn(e);
+                result = ERROR;
+            }
+        } else {
+            result = COURSE_DETAILS_JSP;
+            model.addAttribute(model.addAttribute(TEACHERS, userLogic.getTeachers()));
```

Comment:
Not my code. Actually, I don't remember seeing this before... I might have just missed it during the respective code review.